### PR TITLE
RPE-657 Supplementary Evidence bug fix

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RPE-657

### Change description ###
Updated Attach Scanned Documents Event handler to  append 
the scanned documents to the existing evidence documents in the ccd case.
Added functional test to check if the evidence documents in the ccd case 
are not removed by the Supplementary Evidence event handler.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/AttachDocsToSupplementaryEvidence.java [hunk 1]
```diff
@@ -20,6 +32,31 @@ CaseData mapEnvelopeToCaseDataObject(Envelope envelope) {
...
     }
 
+    @Override
+    public CaseDataContent buildCaseDataContent(StartEventResponse eventResponse, Envelope envelope) {
+
+        List<ScannedDocument> ccdScannedDocuments = getScannedDocumentsForSupplementaryEvidence(
+            eventResponse.getCaseDetails()
+        );
+
+        List<ScannedDocument> scannedDocuments = getScannedDocuments(envelope);
```
Comment (timja) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/AttachDocsToSupplementaryEvidence.java]: What do people think, should this be a set to prevent any duplicates when we add all the documents down a line?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/ScannedDocumentsHelper.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        objectMapper.enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS);
+    }
+
+    private ScannedDocumentsHelper() {
+        // utility class
+    }
+
+    @SuppressWarnings("unchecked")
+    public static List<ScannedDocument> getScannedDocumentsForExceptionRecord(CaseDetails caseDetails) {
+        List<Map<String, Object>> data = (List<Map<String, Object>>) caseDetails.getData().get("scanRecords");
```
Comment (timja) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/ScannedDocumentsHelper.java]: @nitinprabhu do we need different property names in exception records and supplementary evidence (`scanRecords` and `scannedDocuments`) or could we pick one name and go with that.

@aliveni don't do that in this PR though, unrelated change, just would be good to fix it

Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/ScannedDocumentsHelper.java]: @timja I don't think so we need different names. Yes can be clubbed into one 👍

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/AttachDocsToSupplementaryEvidence.java (hunk 1 above)
Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/AttachDocsToSupplementaryEvidence.java]: I think we should not filter any documents even if they are duplicates. Caseworker will need to remove it through UI once they are satisfied that document is duplicate or not required for case progression.
